core/dnsserver/onstartup.go

```diff
@@ -2,14 +2,23 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
+	"github.com/coredns/coredns/plugin/pkg/dnsutil"
 )
 
+// regex1035PrefSyntax() returns regexp-object for RFC1035 preferred syntax.
+func regex1035PrefSyntax() *regexp.Regexp {
+	matchDomain, _ := regexp.Compile("(^([A-Za-z]([A-Za-z0-9-]*[A-Za-z0-9])(\\.([A-Za-z0-9-]+))*\\.)$)|^(\\.)$")
```

Comment:
I think it's cleaner to check for the root domain separately. This will simplify the regex (reducing nested parentheses), and make it easier to read.
Also if you use back tics to quote the string, you don't need to escape the `\.`s.  This will also make it easier to read.